presto-hive/src/main/java/com/facebook/presto/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -578,6 +588,128 @@ public void dropRole(String role)
+                    checkArgument(existingRoles.contains(role), "Role does not exist: %s", role);
+                    if (grantee.getType() == ROLE) {
+                        checkArgument(existingRoles.contains(grantee.getName()), "Role does not exist: %s", grantee.getName());
+                    }
+
+                    RoleGrant grantWithAdminOption = new RoleGrant(grantee, role, true);
+                    RoleGrant grantWithoutAdminOption = new RoleGrant(grantee, role, false);
+
+                    if (withAdminOption) {
+                        modifiedGrants.remove(grantWithoutAdminOption);
+                        modifiedGrants.add(grantWithAdminOption);
+                    }
+                    else {
+                        modifiedGrants.remove(grantWithAdminOption);
+                        modifiedGrants.add(grantWithoutAdminOption);
+                    }
+                }
+            }
+            modifiedGrants = removeDuplicatedEntries(modifiedGrants);
+            if (!existingGrants.equals(modifiedGrants)) {
+                writeRoleGrantsFile(modifiedGrants);
+            }
+        }
+    }
+
+    @Override
+    public void revokeRoles(Set<String> roles, Set<PrestoPrincipal> grantees, boolean adminOptionFor, PrestoPrincipal grantor)
+    {
+        synchronized (rolesLock) {
+            Set<RoleGrant> existingGrants = listRoleGrantsSanitized();
+            Set<RoleGrant> modifiedGrants = new HashSet<>(existingGrants);
+            for (PrestoPrincipal grantee : grantees) {
+                for (String role : roles) {
+                    RoleGrant grantWithAdminOption = new RoleGrant(grantee, role, true);
+                    RoleGrant grantWithoutAdminOption = new RoleGrant(grantee, role, false);
+
+                    if (modifiedGrants.contains(grantWithAdminOption) || modifiedGrants.contains(grantWithoutAdminOption)) {
+                        if (adminOptionFor) {
+                            modifiedGrants.remove(grantWithAdminOption);
+                            modifiedGrants.add(grantWithoutAdminOption);
+                        }
+                        else {
+                            modifiedGrants.remove(grantWithAdminOption);
+                            modifiedGrants.remove(grantWithoutAdminOption);
+                        }
+                    }
+                }
+            }
+            modifiedGrants = removeDuplicatedEntries(modifiedGrants);
+            if (!existingGrants.equals(modifiedGrants)) {
+                writeRoleGrantsFile(modifiedGrants);
+            }
+        }
+    }
+
+    @Override
+    public Set<RoleGrant> listRoleGrants(PrestoPrincipal principal)
+    {
+        synchronized (rolesLock) {
+            return ImmutableSet.copyOf(listRoleGrantsSanitized().stream()
+                    .filter(grant -> grant.getGrantee().equals(principal))
+                    .collect(toSet()));
+        }
+    }
+
+    private Set<RoleGrant> listRoleGrantsSanitized()
```

Comment:
This method is misleading. It suggests is pure getter with some postprocessing but in fact it may change the content of the file (when file is not sanitized). I'd prefer having a separate method for it.